keras_core/layers/preprocessing/random_crop.py

```diff
@@ -66,7 +66,7 @@ def __init__(
 
     def call(self, inputs, training=True):
         inputs = self.backend.cast(inputs, self.compute_dtype)
-        input_shape = self.backend.shape(inputs)
+        input_shape = inputs.shape
         is_batched = len(input_shape) > 3
```

Comment:
We should keep the `backend.shape(inputs)` line unchanged if we want this to work with dynamic inputs. Instead, you can fix the line above by using ` is_batched = len(inputs.shape) > 3`